presto-product-tests/src/main/java/com/facebook/presto/tests/hive/TestHiveMixedCaseSupport.java

```diff
@@ -0,0 +1,218 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.tests.hive;
+
+import io.prestodb.tempto.ProductTest;
+import org.testng.annotations.Test;
+
+import static com.facebook.presto.tests.TestGroups.MIXED_CASE;
+import static io.prestodb.tempto.assertions.QueryAssert.Row.row;
+import static io.prestodb.tempto.assertions.QueryAssert.assertThat;
+import static io.prestodb.tempto.query.QueryExecutor.query;
+
+public class TestHiveMixedCaseSupport
+        extends ProductTest
+{
+    private static final String SCHEMA_NAME = "hivemixedcaseon";
+    private static final String SCHEMA_NAME_UPPER = "HIVEMIXEDCASEON1";
+
+    /*
+     * Cleanup test case for removing schemas and tables created during mixed-case table tests.
+     *
+     * This test ensures that all test schemas and tables are properly dropped to maintain a clean test environment.
+     */
+    @Test(groups = {MIXED_CASE})
+    public void testCleanupMixedCaseTablesAndSchemas()
+    {
+        // Drop tables if they exist
+        query("DROP TABLE IF EXISTS " + SCHEMA_NAME + ".testtable");
+        query("DROP TABLE IF EXISTS " + SCHEMA_NAME + ".testtable0");
+        query("DROP TABLE IF EXISTS " + SCHEMA_NAME + ".TestTable1");
+        query("DROP TABLE IF EXISTS " + SCHEMA_NAME + ".TESTTABLE2");
+        query("DROP TABLE IF EXISTS " + SCHEMA_NAME_UPPER + ".TESTTABLE4");
+        query("DROP TABLE IF EXISTS " + SCHEMA_NAME_UPPER + ".testtable02");
+        query("DROP TABLE IF EXISTS " + SCHEMA_NAME_UPPER + ".TESTTABLE3");
+
+        // Drop schemas if they exist
+        query("DROP SCHEMA IF EXISTS " + SCHEMA_NAME);
+        query("DROP SCHEMA IF EXISTS " + SCHEMA_NAME_UPPER);
+    }
+
+    /*
+     * Test cases for creating schemas with different naming conventions in Hive.
+     *
+     * This class includes tests for:
+     * 1. Creating a schema with a lowercase name.
+     * 2. Creating a schema with a mixed-case name that has the same syllables as an existing schema.
+     * 3. Creating a schema with a mixed-case name.
+     * 4. Creating a schema with an uppercase name.
+     *
+     * Each test ensures the schema is created successfully.
+     */
+    @Test(groups = {MIXED_CASE})
+    public void testCreateSchemasWithMixedCaseNames()
+    {
+        // Define schema names and their expected stored values in Hive
+        String schemaNameMixedSyllables = "HiveMixedCaseOn";
+        String schemaNameMixed = "HiveMixedCase";
+
+        String schemaNameLowerStored = "hivemixedcaseon";
+        String schemaNameMixedSyllablesStored = "hivemixedcaseon";
+        String schemaNameMixedStored = "hivemixedcase";
+        String schemaNameUpperStored = "hivemixedcaseon1";
```

Comment:
Can't we just get this from `SCHEMA_NAME_UPPER.toLowerCase()`? Same with the rest. If we expect them all to be returned with lower case, just call the lowercase method instead during the assertion. It's clearer what the test is expecting